src/slice_file.rs

```diff
@@ -168,6 +169,67 @@ impl SliceFile {
     }
 }
 
+pub trait SliceFileHashable {
+    /// Hashes the SliceFile using a SHA-256 hash and returns the 32-byte array.
+    fn compute_sha256_hash_as_bytes(&self) -> [u8; 32];
+
+    /// Hashes the SliceFile using a SHA-256 hash and returns the hash as a hex string.
+    fn compute_sha256_hash(&self) -> String;
```

Comment:
Both implementations of this are identical, so it's better to just define the function once:
```suggestion
    fn compute_sha256_hash(&self) -> String {
        self.compute_sha256_hash_as_bytes()
            .iter()
            .map(|byte| format!("{:02x}", byte))
            .collect()
    }
```